### PR TITLE
Implement provision watcher patch tests and fix some typo

### PR DESCRIPTION
**Pull request by cherrycl**
1. Implement provision watcher patch tests.
2. Update Deivce to Device.
#291
Signed-off-by: Cherry Wang <cherry@iotechsys.com>

### CODE REVIEW

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot [hunk 1]
```diff
@@ -12,58 +13,116 @@ ${api_version}    v2
...
+    Create Multiple Profiles/Services And Generate Multiple Provision Watchers Sample
+    Create Provision Watcher ${provisionwatcher}
+    ${labels}=  Create List  provision-watcher-example  provision-watcher-update
+    ${update_labels}=  Create Dictionary  name=Test-Provision-Watcher  labels=${labels}  apiVersion=${api_version}
+    ${update_adminstate}=  Create Dictionary  name=Test-Provision-Watcher-Locked  adminState=UNLOCKED  apiVersion=${api_version}
+    ${identifiers}=  Load data file "core-metadata/identifiers.json" and get variable "identifiers"
+    Set To Dictionary  ${identifiers}  address=0.0.0.0  port=123
+    ${update_identifiers}=  Create Dictionary  name=Test-Provision-Watcher-AutoEvents  identifiers=${identifiers}  apiVersion=${api_version}
+    ${ports}=  Create List  111  222  333
+    ${blockingIdentifiers}=  Load data file "core-metadata/identifiers.json" and get variable "blockingIdentifiers"
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot:98]: identifiers.json doesn't exist

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot [hunk 2]
```diff
@@ -12,58 +13,116 @@ ${api_version}    v2
...
-    Given Create Provision Watchers
-    When Update Provision Watchers
+    Given Create Provision Watchers And Generate Multiple Provision Watchers Sample For Updating Data
+    When Update Provision Watchers ${provisionwatcher}
     Then Should Return Status Code "207"
     And Should Return Content-Type "application/json"
     And Item Index All Should Contain Status Code "200"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
     And Provision Watcher Data Should Be Updated
+    #[Teardown]  Delete Multiple Devices Sample, Profiles Sample And Services Sample
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot:23]: Should this be unmarked?

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot [hunk 3]
```diff
@@ -12,58 +13,116 @@ ${api_version}    v2
...
+    Then Should Return Status Code "400"
     And Should Return Content-Type "application/json"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete Multiple Provision Watchers Sample, Profiles Sample And Services Sample
 
 ErrProWatcherPATCH004 - Update provision watcher with autoEvents but no frequency
-    Given Create Provision Watchers
-    When Update Provision Watchers
-    Then Should Return Status Code "404"
+    ${autoEvents}=  Set autoEvents values  24h  false  DeviceValue_Boolean_RW
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot:56]:
```suggestion
    ${autoEvents}=  Set autoEvents values  ${EMPTY}  false  DeviceValue_Boolean_RW
```
And remove https://github.com/edgexfoundry/edgex-taf/blob/42b0bbb31671a490b1ac7acc448cd950b10485ff/TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot#L59
Would this be better?

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot [hunk 4]
```diff
@@ -12,58 +13,116 @@ ${api_version}    v2
...
+    Then Should Return Status Code "400"
     And Should Return Content-Type "application/json"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete Multiple Provision Watchers Sample, Profiles Sample And Services Sample
 
 ErrProWatcherPATCH005 - Update provision watcher with autoEvents but no resource
-    Given Create Provision Watchers
-    When Update Provision Watchers
-    Then Should Return Status Code "404"
+    ${autoEvents}=  Set autoEvents values  24h  false  DeviceValue_Boolean_RW
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot:67]:
```suggestion
    ${autoEvents}=  Set autoEvents values  24h  false  ${EMPTY}
```
And remove https://github.com/edgexfoundry/edgex-taf/blob/42b0bbb31671a490b1ac7acc448cd950b10485ff/TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot#L70

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot (hunk 1 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot:96]: Added identifiers.json.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot (hunk 3 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot]: Updated.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot (hunk 4 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/provisionwatcher/PATCH.robot]: Updated.

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**